module/solutions/helper.js

```diff
@@ -1154,6 +1154,256 @@ module.exports = class SolutionsHelper {
+              isAPrivateProgram: false,
+              author : userId,
+          },{
+            $set : {
+              isDeleted : true
+            }
+          });
+
+          let reponseMessage = null;
+          let result = {};
+          if(solutionData){
+            if(solutionData.isDeleted){
+              reponseMessage = messageConstants.apiResponses.SOLUTION_DELETED;
+              result = solutionData._id;
+            }else{
+              reponseMessage = messageConstants.apiResponses.SOLUTION_CANT_DELETE;
+            }
+            
+          }else{
+              reponseMessage = messageConstants.apiResponses.SOLUTION_CANT_DELETE;
+          }
+
+          return resolve({
+              message: reponseMessage,
+              result: result
+          });
+
+        } catch(error) {
+          return reject(error);
+        }
+      })
+    }
+
+    /**
+     * Move To Trash.
+     * @method
+     * @name moveToTrash
+     * @param {String} solutionId - solution External id.
+     * @param {String} userId - UserId.
+     * @returns {Object} Solution .
+     */
+
+    static moveToTrash(
+      solutionId,
+      userId
+    ) {
+      return new Promise(async (resolve, reject) => {
+        try {
+
+          if(!solutionId) {
+            return resolve({
+                message: messageConstants.apiResponses.SOLUTION_ID_REQUIRED,
+                result: result
+            });
+          }
+
+          let solutionData = await database.models.solutions.findOneAndUpdate({
+              externalId: solutionId,
+              isAPrivateProgram: false,
+              author : userId,
+              isDeleted:true
+          },{
+            $set : {
+              status : messageConstants.common.INACTIVE_STATUS,
+            }
+          });
+
+          if(solutionData){
+            var reponseMessage = messageConstants.apiResponses.SOLUTION_MOVED_TO_TRASH;
+            var result = solutionData._id;
+          }else{
+            var reponseMessage = messageConstants.apiResponses.SOLUTION_CANT_DELETE;
+            var result = {};
+          }
+
+          return resolve({
+              message: reponseMessage,
+              result: result
+          });
+
+        } catch(error) {
+          return reject(error);
+        }
+      })
+    }
+
+     /**
+     * Restore From Trash.
+     * @method
+     * @name restoreFromTrash
+     * @param {String} solutionId - solution External id.
+     * @param {String} userId - UserId.
+     * @returns {Object} Solution .
+     */
+
+    static restoreFromTrash(
+      solutionId,
+      userId
+    ) {
+      return new Promise(async (resolve, reject) => {
+        try {
+
+          if(!solutionId) {
+            return resolve({
+                message: messageConstants.apiResponses.SOLUTION_ID_REQUIRED,
+                result: result
+            });
+          }
+
+          let solutionData = await database.models.solutions.findOneAndUpdate({
```

Comment:
@priyanka-TL - Use update query as shared in slack and isAPrivateProgram has to be true